integration_tests/sdk/aqueduct_tests/lazy_execution_test.py

```diff
@@ -238,3 +239,21 @@ def copy_field(df):
     data_validator.check_saved_artifact_data(
         flow, review_copied.id(), expected_data=copy_field.local(reviews)
     )
+
+
+def test_eager_then_lazy(client):
+    """For an operator, checks that we can switch its execution mode from eager to lazy."""
+
+    @op
+    def number():
+        return 1
+
+    result = number()
+    assert isinstance(result, NumericArtifact)
+    assert result.get() == 1
+
+    global_config({"lazy": True})
+
+    result = number()
+    assert isinstance(result, GenericArtifact)
```

Comment:
btw the idea is that the data field isn't set on this GenericArtifact right? Until get() is called.

Reply:
if the mode is lazy, `result` will be of type `GenericArtifact` because we don't know the exact type yet. This won't be the case when we're evaluating it eagerly.